### PR TITLE
Add build info to logs

### PR DESCRIPTION
**Pull request by ronensc**
# Issue link

# What changes have been made

This PR adds a log line with the build version and build date of the CFO.
Knowing the build version and build date is useful when debugging historical logs and correlating them with the actual code that produced them.

build version format:
```
<tag>[-<sha>][-dirty]
```

`<tag>`: The name of the most recent tag (stripping the prefix 'v')
`<sha>`: If HEAD is not pointed to a tag, then the hash of HEAD is added.
`dirty`: If the repository contains any changes, "-dirty" is added.

Note: Once approved, I think it will be beneficial to add a similar PR to MCAD and will be happy to do that.

# Verification steps

Build and run:
```
make build && ./bin/manager
```

The first log line should look like:
```
2023-11-13T17:27:45+02:00       INFO    setup   Build info      {"version": "1.0.0-7991ad5", "date": "2023-11-13 17:27"}
```

Running without `make build` should yield:
```
$ go run main.go
2023-11-13T17:53:52+02:00       INFO    setup   Build info      {"version": "UNKNOWN", "date": "UNKNOWN"}
```

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by ronensc** [comment]
cc @eranra @rachelt44

File: main.go [hunk 1]
```diff
@@ -85,6 +87,7 @@ func main() {
 	zapOptions.BindFlags(flag.CommandLine)
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&zapOptions)))
+	setupLog.Info("Build info", "version", BuildVersion, "date", BuildDate)
```
Comment (astefanutti) [inline main.go:90]: I'd also be useful to log the MCAD and InstaScale versions, sourced from the Makefile as well.

Comment (ronensc) [reply main.go]: done

**Pull request merged**